### PR TITLE
fix(django): Support for Django 3.1

### LINKED ISSUE #704: Convert django app to ASGI
I converted my app to ASGI but then the Sentry integration stopped working like here:

`TypeError: sentry_patched_load_middleware() got an unexpected keyword argument 'is_async'`

If I comment out the Sentry integration `uvicorn` runs th eapp just fine.

### PR DESCRIPTION
**Pull request by untitaker**
* Django 3.1a1 adds more parameters to `load_middleware` which we do not really care about.

* Django 3.1a1 starts executing exception handlers in a random thread/with the wrong context. Turns out they have their own implementation of context local that is necessary to be able to find the right hub. See also https://github.com/getsentry/sentry-docs/pull/1721

More support is required for supporting async middlewares once Django 3.1 comes out but this should unbreak basic usage of the sdk.

Fix #704

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -73,10 +73,10 @@ envlist =
 deps =
     -r test-requirements.txt
 
-    django-{1.11,2.0,2.1,2.2,3.0}: djangorestframework>=3.0.0,<4.0.0
-    py3.7-django-{1.11,2.0,2.1,2.2,3.0}: channels>2
-    py3.7-django-{1.11,2.0,2.1,2.2,3.0}: pytest-asyncio==0.10.0
-    {py2.7,py3.7}-django-{1.11,2.2,3.0}: psycopg2-binary
+    django-{1.11,2.0,2.1,2.2,3.0,dev}: djangorestframework>=3.0.0,<4.0.0
+    {py3.7,py3.8}-django-{1.11,2.0,2.1,2.2,3.0,dev}: channels>2
+    {py3.8,py3.7}-django-{1.11,2.0,2.1,2.2,3.0,dev}: pytest-asyncio==0.10.0
```
Comment (rhcarvalho) [inline tox.ini:78]:
```suggestion
    {py3.7,py3.8}-django-{1.11,2.0,2.1,2.2,3.0,dev}: pytest-asyncio==0.10.0
```

Keep them sorted for our sanity?

File: tests/integrations/django/asgi/test_asgi.py [hunk 2]
```diff
@@ -18,8 +18,11 @@
 
 @pytest.mark.parametrize("application", APPS)
 @pytest.mark.asyncio
-async def test_basic(sentry_init, capture_events, application):
+async def test_basic(sentry_init, capture_events, application, request):
     sentry_init(integrations=[DjangoIntegration()], send_default_pii=True)
+    Hub.main.bind_client(Hub.current.client)
+    request.addfinalizer(lambda: Hub.main.bind_client(None))
```
Comment (rhcarvalho) [inline tests/integrations/django/asgi/test_asgi.py:24]: Why do we need this now?

File: sentry_sdk/integrations/django/__init__.py [hunk 3]
```diff
@@ -337,6 +307,54 @@ def _patch_django_asgi_handler():
...
+
+        scope.add_event_processor(
+            _make_event_processor(weakref.ref(request), integration)
+        )
+
+
+def _patch_get_response():
+    # type: () -> None
+    """
+    patch get_response, because at that point wea have the Django request object
```
Comment (rhcarvalho) [inline sentry_sdk/integrations/django/__init__.py:339]:
```suggestion
    patch get_response, because at that point we have the Django request object
```

**Comment by untitaker** [comment]
>Do we have tests covering all the monkey-patching?

We have tests exercising a demo Django app in a way that hits the codepaths we're monkeypatching.

**Comment by untitaker** [comment]
For anybody wondering this is blocked on https://github.com/django/asgiref/issues/170, ideally we'd figure out the bug in Django

File: tests/integrations/django/asgi/test_asgi.py (hunk 2 above)
Comment (untitaker) [reply tests/integrations/django/asgi/test_asgi.py]: I had an argument one week ago for this, but can't remember it. I tried removing those lines and it works just fine. Removed.

**Pull request merged**